backend/api/v1/idp_service.go

```diff
@@ -221,6 +221,47 @@ func (s *IdentityProviderService) TestIdentityProvider(ctx context.Context, requ
 		if userInfo == nil {
 			return nil, status.Errorf(codes.InvalidArgument, "missing user info")
 		}
+	} else if identityProvider.Type == v1pb.IdentityProviderType_OIDC {
+		// Find client secret for those existed identity providers.
+		if request.IdentityProvider.Config.GetOidcConfig().ClientSecret == "" {
+			storedIdentityProvider, err := s.getIdentityProviderMessage(ctx, request.IdentityProvider.Name)
+			if err != nil {
+				return nil, status.Errorf(codes.Internal, "failed to find identity provider, error: %s", err.Error())
+			}
+			if storedIdentityProvider != nil {
+				request.IdentityProvider.Config.GetOidcConfig().ClientSecret = storedIdentityProvider.Config.GetOidcConfig().ClientSecret
+			}
```

Comment:
Two questions:

1. Why do we need to check `nil`-ness again after we've checked the error? Can't we just assume we got a valid value if the error is `nil`?
2. What happens with the IdP is `nil`? Right now we just continue execution but it doesn't seems to make sense to continue.

Reply:
Since we don't return the `client_secret` to frontend, we need to go get IdP in database for those created IdP. And the result we get may be nil.

https://github.com/bytebase/bytebase/blob/main/backend/store/idp.go#L137

Reply:
OK.. in that case, we still need to return an error here instead of continuing, wdyt?

Reply:
SGTM